dali/test/python/test_operator_readers_webdataset_corner.py

```diff
@@ -0,0 +1,378 @@
+
+    def earlier_final_offset_error():
+        index_file = tempfile.NamedTemporaryFile()
+        index_file.write(b"512 2\n0 jpg 0\n1024 png 512")
+        index_file.flush()
+        webdataset_pipeline = webdataset_raw_pipeline(
+            get_dali_extra_path() + "/db/webdataset/MNIST/devel-0.tar",
+            index_file.name,
+            ["jpg", "cls"],
+            batch_size=1,
+            device_id=0,
+            num_threads=1,
+        )
+        webdataset_pipeline.build()
+
+    assert_raises(RuntimeError, earlier_final_offset_error)
+
+    def offset_not_divisible_error():
+        index_file = tempfile.NamedTemporaryFile()
+        index_file.write(b"3072 2\n0 jpg 0\n1030 png 512")
+        index_file.flush()
+        webdataset_pipeline = webdataset_raw_pipeline(
+            get_dali_extra_path() + "/db/webdataset/sample-tar/types_contents.tar",
+            index_file.name,
+            "float16",
+            batch_size=1,
+            device_id=0,
+            num_threads=1,
+        )
+        webdataset_pipeline.build()
+
+    assert_raises(RuntimeError, offset_not_divisible_error)
+
+    def final_offset_not_divisible_error():
+        index_file = tempfile.NamedTemporaryFile()
+        index_file.write(b"3100 2\n0 jpg 0\n1024 png 512")
+        index_file.flush()
+        webdataset_pipeline = webdataset_raw_pipeline(
+            get_dali_extra_path() + "/db/webdataset/sample-tar/types_contents.tar",
+            index_file.name,
+            "float16",
+            batch_size=1,
+            device_id=0,
+            num_threads=1,
+        )
+        webdataset_pipeline.build()
+
+    assert_raises(RuntimeError, offset_not_divisible_error)
+
+    def offset_order_error():
+        index_file = tempfile.NamedTemporaryFile()
+        index_file.write(b"3072 2\n1024 png 0\n0 jpg 512")
+        index_file.flush()
+        webdataset_pipeline = webdataset_raw_pipeline(
+            get_dali_extra_path() + "/db/webdataset/sample-tar/types_contents.tar",
+            index_file.name,
+            "float16",
+            batch_size=1,
+            device_id=0,
+            num_threads=1,
+        )
+        webdataset_pipeline.build()
+
+    assert_raises(RuntimeError, offset_order_error)
+
+    def reported_component_sizes_error():
+        index_file = tempfile.NamedTemporaryFile()
+        index_file.write(b"3072 2\n0 int32 40 float64 80 float16 20")
+        index_file.flush()
+        webdataset_pipeline = webdataset_raw_pipeline(
+            get_dali_extra_path() + "/db/webdataset/sample-tar/types_contents.tar",
+            index_file.name,
+            "float16",
+            dtypes=dali.types.FLOAT64,
+            batch_size=1,
+            device_id=0,
+            num_threads=1,
+        )
+        webdataset_pipeline.build()
+        webdataset_pipeline.run()
+        webdataset_pipeline.run()
+    assert_raises(RuntimeError, reported_component_sizes_error)
```

Comment:
Again, please check error message using nose_utils.

Reply:
done, will be pushed soon